### PR TITLE
VTOL: fix roll inversion

### LINKED ISSUE #4797: Check roll direction for VTOL
I have the odd symptom that the ailerons signs are inverted for VTOL in SITL.

### PR DESCRIPTION
**Pull request by jlecoeur**
fixes #4797 

**Describe problem solved by the proposed pull request**
The VTOL attitude controller generates reversed fixed-wing roll commands. As a consequence, all VTOL mixers currently have a minus sign on roll actuators to compensate.

**Describe your preferred solution**
- Fix sign of the fixed wing commands generated by the VTOL attitude controller
- Fix builtin mixers
- Add a parameter to force users of custom VTOL mixers to check roll control surfaces before flight
- Fix builtin airframes scripts to bypass the new parameter

**Flight testing**
This requires extensive flight testing on all possible VTOL platforms.

Test card 1:
- Flash firmware on VTOL
- **Before flight**: Check direction of roll control surfaces
- Fly standard mission, the behaviour should be unchanged

Test card 2:
- set parameter V19_VT_ROLLDIR to 1
- try to arm
- the system should not generate actuator commands, error messages should be displayed on QGC.
- set parameter V19_VT_ROLLDIR to 0
- arm properly

### CODE REVIEW

**Comment by jlecoeur** [comment]
I see spurious jenkins errors like
```
[21/667] Copying NuttX/nuttx to build/px4_fmu-v2_default/NuttX
FAILED: NuttX/nuttx_copy.stamp 
cd /tmp/jenkins/workspace/e-compile_pr-vtol_roll_inversion && cp -aRfl platforms/nuttx/NuttX/nuttx build/px4_fmu-v2_default/NuttX && cmake -E touch /tmp/jenkins/workspace/e-compile_pr-vtol_roll_inversion/build/px4_fmu-v2_default/NuttX/nuttx_copy.stamp
cp: cannot stat 'platforms/nuttx/NuttX/nuttx/wireless/bluetooth/bt_smp.h': No such file or directory
ninja: build stopped: subcommand failed.
Makefile:171: recipe for target 'px4_fmu-v2_default' failed
make: *** [px4_fmu-v2_default] Error 1
script returned exit code 2
```

File: src/modules/vtol_att_control/vtol_att_control_params.c [hunk 1]
```diff
@@ -314,3 +314,26 @@ PARAM_DEFINE_INT32(VT_FW_DIFTHR_EN, 0);
...
+ * - Standard VTOL platforms should be unaffected and this parameter can be ignored.
+ * - Custom VTOL platforms may crash if no action is taken, please check the direction of
+ * deflection of roll control surfaces before flight. Fix the roll mixer if necessary.
+ *
+ * Set to 1 to disable VTOL actuator outputs and display an info message (default).
+ * Set to 0 AFTER CAREFULLY CHECKING the direction of deflection of roll control surfaces.
+ *
+ * @min 0
+ * @max 1
+ * @decimal 0
```
[the remark this answers is not in the document]
Comment (jlecoeur) [reply src/modules/vtol_att_control/vtol_att_control_params.c]: Ok
BTW, what is the difference?

[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/vtol_att_control/vtol_att_control_params.c]: It's grouped in QGC.

**Comment by jlecoeur** [comment]
I rebased and CI now passes, thanks @dagar

File: src/modules/vtol_att_control/vtol_att_control_main.cpp [hunk 2]
```diff
@@ -663,7 +668,19 @@ void VtolAttitudeControl::task_main()
 			_vtol_type->update_transition_state();
 		}
 
-		_vtol_type->fill_actuator_outputs();
+		if (_params.v19_vt_rolldir) {
+
+			// The mixer may not have been adapted to the roll inversion in v1.9
+			// Display error message and do not fill actuator outputs
+			// TODO: remove the parameter and this error message in v1.10
+			mavlink_log_critical(&_mavlink_log_pub,
```
Comment (dagar) [inline src/modules/vtol_att_control/vtol_att_control_main.cpp]: We can't spam the message this quickly.

Comment (jlecoeur) [reply src/modules/vtol_att_control/vtol_att_control_main.cpp]: Ok, I added to the ugliness with a static counter to throttle down the msg rate.

Comment (dagar) [reply src/modules/vtol_att_control/vtol_att_control_main.cpp]: We can squash the v19 bits into a single commit to revert later this year.

Comment (jlecoeur) [reply src/modules/vtol_att_control/vtol_att_control_main.cpp]: done

**Comment by dagar** [comment]
So far looks good. Let's get organized to test everything exhaustively.

**Comment by jlecoeur** [comment]
@Tony3dr  any update on the flight tests?

**Comment by Tony3dr** [comment]
> @Tony3dr any update on the flight tests?

Sorry for the delay, we are expecting replacement frames for the VTOL we have. We should get the replacements this week.

**Comment by jlecoeur** [comment]
Great!

**Comment by Junkim3DR** [comment]
@jlecoeur we were unable to download the firmware from Jenkins.
"Compile All Boards — The build of this commit was aborted"

**Comment by jlecoeur** [comment]
@Junkim3DR it should be good now

**Comment by dannyfpv** [comment]
tested on pixhawk4 mini v5 converge frame good convertion to plane mode
https://review.px4.io/plot_app?log=da0a2356-eb30-4b77-8deb-696798a263ba

**Comment by dagar** [comment]
Thank you for finally addressing this @jlecoeur

**Comment by dagar** [comment]
Follow Up issue https://github.com/PX4/Firmware/issues/11650.

**Pull request merged**